src/api/app/controllers/staging/staged_requests_controller.rb

```diff
@@ -1,5 +1,81 @@
-class Staging::StagedRequestsController < Staging::ProjectsController
+class Staging::StagedRequestsController < ApplicationController
+  before_action :require_login, except: [:index]
+  before_action :set_staging_project
+  before_action :set_staging_workflow, :set_project, :set_xml_hash, only: [:create, :destroy]
+
   def index
-    @requests = @project.staged_requests
+    @requests = @staging_project.staged_requests
+  end
+
+  def create
+    authorize @staging_project, :update?
+    requests = @staging_workflow.unassigned_requests.where(number: request_numbers)
+
+    result = ::Staging::StageRequests.new(requests: requests, staging_project_name: @staging_project.name).perform
+    @staging_project.staged_requests << result
+    unassigned_requests = request_numbers - result.pluck(:number).map(&:to_s)
+
+    if unassigned_requests.empty?
+      render_ok
+    else
+      render_error(
+        status: 400,
+        errorcode: 'invalid_request',
+        message: "Could not assign requests #{unassigned_requests.join(', ')} to #{@staging_project}."
+      )
+    end
+  end
+
+  def destroy
+    authorize @staging_project, :update?
+    requests = @staging_project.staged_requests.where(number: request_numbers)
+    package_names = requests.joins(:bs_request_actions).pluck('bs_request_actions.target_package')
+
+    @staging_project.staged_requests.delete(requests)
+    not_unassigned_requests = request_numbers - requests.pluck(:number).map(&:to_s)
+
+    result = @staging_project.packages.where(name: package_names).destroy_all
+    not_deleted_packages = package_names - result.pluck(:name)
+
+    if not_unassigned_requests.empty? && not_deleted_packages.empty?
+      render_ok
+    else
+      message = 'Error while unassigning requests: '
+      message << "Requests with number #{not_unassigned_requests.join(', ')} not found. " unless not_unassigned_requests.empty?
+      message << "Could not delete packages #{not_deleted_packages.join(', ')}." unless not_deleted_packages.empty?
+      render_error(
+        status: 400,
+        errorcode: 'invalid_request',
+        message: message
+      )
+    end
+  end
+
+  private
+
+  def set_xml_hash
+    @xml_hash = (Xmlhash.parse(request.body.read) || {}).with_indifferent_access
+  end
+
+  def request_numbers
+    [@xml_hash[:number]].flatten
+  end
+
+  def set_staging_project
+    @staging_project = Staging::StagingProject.find_by!(name: params[:staging_project_name])
+  end
+
+  def set_staging_workflow
+    @staging_workflow = @staging_project.staging_workflow
+    return if @staging_workflow
+    render_error(
+      status: 422,
+      errorcode: 'invalid_request',
+      message: "#{@staging_project} is not a valid staging project, can't assign requests to it."
+    )
+  end
+
+  def set_project
```

Comment:
where is this used? :confused:

Reply:
In the before_filters here: https://github.com/openSUSE/open-build-service/pull/6190/files#diff-95312578b7db895b4a67e567d2005e54R4